snode-src/duckcoin.go

```diff
@@ -412,29 +421,34 @@ func makeSignature(privkey string, message string) (string, error) {
 	return b64(data), nil
 }
 
+// The calculateHash function is used to calculate the hash of a block, this is used in the creation of the blockchain
 func calculateHash(block Block) string {
 	block.Hash = ""
 	block.Tx.Signature = ""
-	return shasum([]byte(toJson(block)))
+	return shasum([]byte(toJSON(block)))
 }
 
+// The shasum function calculates the hash of an array of bytes
 func shasum(record []byte) string {
 	h := sha256.New()
 	h.Write(record)
 	hashed := h.Sum(nil)
 	return hex.EncodeToString(hashed)
 }
 
+// isHashSolution checks if the hash of a block is a valid hash for the given difficulty
 func isHashSolution(hash string) bool {
 	prefix := strings.Repeat("0", Difficulty)
 	return strings.HasPrefix(hash, prefix)
 }
 
-func toJson(v interface{}) string {
+// The toJSON function converts an object to a JSON string
+func toJSON(v interface{}) string {
 	s, _ := json.MarshalIndent(v, "", "   ")
 	return string(s)
 }
 
+// The argsHaveOption function checks if the command line arguments have the given option
```

Comment:
I would argue that documenting this function is not really necessary, but if you really want to, include this:
```suggestion
// argsHaveOption checks if the command line arguments have the given option. It takes in the short and the long forms of the option. Example: argsHaveOption("help", "h")
```
```suggestion
```